bundle/src/main/java/com/adobe/acs/commons/workflow/process/impl/SetImageOrientationProcess.java

```diff
@@ -0,0 +1,255 @@
+
+                    String tagId = getOrientation(assetDetails, config);
+
+                    log.debug("Orientation tag is {}", tagId);
+
+                    if (tagId != null) {
+                        Tag tag = tagManager.resolve(tagId);
+                        if (tag != null) {
+                            addTagToResource(assetResource, tag, tagManager);
+                            log.debug("Orientation tag set");
+                        } else {
+                            log.warn("Unable to resolve tag {} - check configuration for Set Image Orientation workflow step", tagId);
+                        }
+                    } else {
+                        log.warn("Unable to set tag.");
+                    }
+                }
+
+            }
+
+        } catch (RepositoryException re) {
+            log.error("Repository Exception", re);
+        }
+
+    }
+
+    private void addTagToResource(Resource resource, Tag tag, TagManager tagManager) {
+        Resource metadataResource = getMetadataResource(resource);
+        Tag[] currentTags = tagManager.getTags(metadataResource);
+        Tag[] updatedTags = (Tag[]) ArrayUtils.add(currentTags, tag);
+        tagManager.setTags(metadataResource, updatedTags, true);
+
+    }
+
+
+    private Resource getMetadataResource(Resource resource) {
+        return resource.getChild(JcrConstants.JCR_CONTENT + "/" + DamConstants.METADATA_FOLDER);
+    }
+
+    protected String getOrientation(AssetDetails assetDetails, Configuration config) {
+
+        long width = 0;
+        long height = 0;
+        try {
+            width = assetDetails.getWidth();
+            height = assetDetails.getHeight();
+        } catch (RepositoryException re) {
+            log.warn("Unable to get width / height for asset: {}", assetDetails.getAsset().getPath());
+        }
+
+        log.debug("Width {} Height {}", width, height);
+
+        String tagId = null;
+
+        if (width > 0 && height > 0) {
+            float ratio = (float) width / height;
+
+            log.debug("Ratio is {}", ratio);
+
+            tagId = getTagId(config, ratio);
+        }
+
+        // Asset doesn't have width/height so we can't decide correct orientation
+        if (tagId == null) {
+            log.debug("Couldn't calculate orientation");
+        }
+
+        return tagId;
+    }
+
+    private String getTagId(Configuration config, float ratio) {
+        for (ConfigRule rule : config.getConfig()) {
+            if (rule.operator.equals(GT) && ratio > rule.limit) {
+                return rule.tagId;
+            }
+            if (rule.operator.equals(LT) && ratio < rule.limit) {
+                return rule.tagId;
+            }
+
+            if (rule.operator.equals(DEFAULT)) {
+                return rule.tagId;
+            }
+
+        }
+        return null;
+    }
+
+
+    /**
+     * Inner class for parsing and storing workflow step configuration
+     */
+    static class Configuration {
```

Comment:
Is there a reason these 2 inner classes are declared as static? Looking at the code that doesn't seem to be necessary, but perhaps I'm missing something.

Reply:
I was thinking that there's only single configuration at any time but agree that this makes more sense without being static. Updated.